src/Host/Broker/Impl/Security/SecurityManager.cs

```diff
@@ -49,32 +51,35 @@ public class SecurityManager {
             return new ClaimsPrincipal(identity);
         }
 
-        private uint CreateProfile(string username, string domain, string password,out StringBuilder profilePath, out bool profileExists) {
-            uint error = 0;
-            profilePath = new StringBuilder();
-            profileExists = false;
-            using (NamedPipeClientStream client = new NamedPipeClientStream("RUserCreatorPipe"))
-            using (BinaryReader reader = new BinaryReader(client))
-            using (BinaryWriter writer = new BinaryWriter(client)) {
+        private async Task<JArray> CreateProfileAsync(string username, string domain, string password, CancellationToken ct) {
```

Comment:
Given that all inputs are separate strongly typed values, the return should also be that, rather than `JArray` - i.e. this method should take responsibility both for serialization of the request, and for deserialization of the result.